### PR TITLE
Merge UserCollection into Master

### PR DESCRIPTION
**Pull request by prabalsingh24**
### Problem
Merge UserCollection into Master

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,8 +43,10 @@
     "@fortawesome/free-brands-svg-icons": "^5.14.0",
     "@fortawesome/free-solid-svg-icons": "^5.14.0",
     "@fortawesome/react-fontawesome": "^0.1.11",
+    "bluebird": "^3.5.1",
```
Comment (MonkeyDo) [inline package.json:46]: The need for the bluebird library has been removed in a previous PR, and we shouldn't need the package anymore.

File: src/server/helpers/middleware.js [hunk 2]
```diff
@@ -23,6 +23,8 @@ import * as commonUtils from '../../common/helpers/utils';
 import * as error from '../../common/helpers/error';
 import * as utils from '../helpers/utils';
 import type {$Request, $Response, NextFunction} from 'express';
+import Promise from 'bluebird';
```
Comment (MonkeyDo) [inline src/server/helpers/middleware.js:26]: Same comment as on package.json, bluebird has been removed and this line shouldn't be needed.

File: src/client/helpers/setupIconLibrary.js [hunk 3]
```diff
@@ -38,6 +38,9 @@ config.autoAddCss = true;
 // Add Icons to FontAwesome library
 library.add(
 	fab, faAngleDoubleLeft, faAngleDoubleUp, faBook, faCalendarAlt, faChartLine, faCheck, faCircle,
+	faCircleNotch, faCodeBranch, faComment, faCommentDots, faComments, faEnvelope, faExclamationTriangle,
+	faExternalLinkAlt, faGlobe, faGripVertical, faHistory, faInfo, faListUl, faPenNib, faPencilAlt, faPencilRuler,
+	faPlus, faQuestionCircle, faRemoveFormat, faSave, faSearch, faSignInAlt,
```
Comment (MonkeyDo) [inline src/client/helpers/setupIconLibrary.js:43]: It looks like these lines were duplicated during the merge from master.
You'll need to make sure the result does have the 3 new icons in there (faComment, faGripVertical and faSave)

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
I see a couple of merge issues, but nothing dramatic.
After that, it's ready to merge ! 🎉 🚀

**Comment by prabalsingh24** [comment]
Coverage decreased (-1.7%), that's sad. I was hoping it to be increased. But I think because we don't have tests for front end stuff, and alot of front end code was added, it decreased.

**Review by MonkeyDo — APPROVED** [review]
Let's go !

**Pull request merged**